addons/misra.py

```diff
@@ -740,8 +762,33 @@ def getForLoopCounterVariables(forToken):
                     vars_modified.add(tn.variable)
                 elif tn.previous and tn.previous.str in ('++', '--'):
                     vars_modified.add(tn.variable)
-        if cur_clause == 1 and tn.isAssignmentOp and tn.astOperand1.variable:
-            vars_initialized.add(tn.astOperand1.variable)
+        if cur_clause == 1 and tn.isAssignmentOp:
+            var_token = tn.astOperand1
+            while var_token and var_token.str == '.':
+                var_token = var_token.astOperand2
+            if var_token and var_token.variable:
+                vars_initialized.add(var_token.variable)
+        if cur_clause == 1 and tn.function:
+            function_args_in_init = getArguments(tn.next);
+            function_args = tn.function.argument
+            function_scope = get_function_scope(cfg, tn.function)
+            for key, args in function_args.items():
+                if args.nameToken and args.nameToken.valueType and args.nameToken.valueType.pointer > 0:
+                    function_body_start = None
+                    function_body_end = None
+                    if function_scope:
+                        function_body_start = function_scope.bodyStart
+                        function_body_end = function_scope.bodyEnd
+                    if function_scope is None or \
+                            (function_scope is not None and
+                             is_variable_changed(function_body_start, function_body_end, args)):
+                        arg_tok = function_args_in_init[int(key) - 1]
+                        if arg_tok.variable:
+                            vars_initialized.add(arg_tok.variable)
+                        elif arg_tok.isOp and arg_tok.valueType and arg_tok.valueType.pointer > 0 \
```

Comment:
I wonder what you think about `arg_tok.isUnaryOp('&')` instead of `arg_tok.isOp and arg_tok.valueType and arg_tok.valueType.pointer > 0`